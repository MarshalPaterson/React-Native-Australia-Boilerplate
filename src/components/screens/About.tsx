import React, { useContext } from "react";
import { Text } from "react-native";
import SafeAreaView from "react-native-safe-area-view";
import StoreContext from "../../contexts/StoreContext";
import { styles } from "../../styles/AppStyles";

function About(): React.ReactElement {
  const store = useContext(StoreContext);
  return (
    <SafeAreaView style={styles.container}>
      <Text style={styles.title}>{store.title}</Text>
      <Text style={styles.text}>{store.about}</Text>
      <Text style={styles.version}>{store.version}</Text>
    </SafeAreaView>
  );
}

export default About;
