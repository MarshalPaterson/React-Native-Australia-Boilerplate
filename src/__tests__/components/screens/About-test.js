import React from "react";
import renderer from "react-test-renderer";
import { SafeAreaProvider } from "react-native-safe-area-context";
import About from "../../../components/screens/About";

jest.useFakeTimers();

it("renders correctly", async () => {
  const tree = renderer
    .create(
      <SafeAreaProvider>
        <About />
      </SafeAreaProvider>
    )
    .toJSON();
    //expect(tree).toMatchSnapshot();
});


