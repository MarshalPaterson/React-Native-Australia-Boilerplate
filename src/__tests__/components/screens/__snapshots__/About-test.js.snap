// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RNCSafeAreaView
  onInsetsChange={[Function]}
  style={
    Object {
      "flex": 1,
    }
  }
/>
`;
